### PR TITLE
Added "Previous" and "Next" buttons

### LINKED ISSUE #23: Broken Internal Navigation:
### Summary

some pages may not have good navigation, so ill improve the navigation to make more convenient for learners  

### Location

_No response_

### Steps to reproduce

i checked the website i found that navigation isn't that good .

### Expected result

_No response_

### Actual result

_No response_

### Relevant log output (optional)

```text

```

### Screenshots / recordings (optional)

_No response_

### Operating system

Windows

### Browser

Chrome

### Device

Laptop

### Component

_No response_

### Additional context (optional)

_No response_

### Checklist

- [x] I searched existing issues and didn’t find a duplicate
- [ ] I’ve provided steps to reproduce or a minimal example
- [ ] I’ve removed sensitive data from logs/screenshots

### PR DESCRIPTION
**Pull request by haleema-khatun**
As i mentioned in the Issue , i have created "Previous" and "Next" buttons for better navigation.

Users does not need to go to the main menu to backtrack, then can do this with just on click.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-site_description: Comprehensive educational resource for NSW HSC Software Engineering syllabus
+site_name: NSW HSC Software Engineering Textbook
+site_description: Interactive textbook for NSW HSC Software Engineering 11-12 Syllabus
 docs_dir: docs
-repo_url: https://github.com/Eatham532/Software-Engineering-HSC-Textbook/
-site_url: https://eatham532.github.io/Software-Engineering-HSC-Textbook/
-repo_name: Eatham532/Software-Engineering-HSC-Textbook
+repo_url: https://github.com/haleema-khatun/Software-Engineering-HSC-Textbook/
+site_url: https://haleema-khatun.github.io/Software-Engineering-HSC-Textbook/
+repo_name: haleema-khatun/Software-Engineering-HSC-Textbook
```
Comment (Eatham532) [inline mkdocs.yml:6]: You seemed to have changed the repo url to your fork???

File: mkdocs.yml [hunk 2]
```diff
@@ -234,5 +260,168 @@ extra:
           data: 0
           note: >- 
             Thanks for your feedback! Help us improve this page by creating an
-            <a href="https://github.com/Eatham532/Software-Engineering-HSC-Textbook/issues/new/choose" target="_blank" rel="noopener">issue</a>
+            <a href="https://github.com/haleema-khatun/Software-Engineering-HSC-Textbook/issues/new/choose" target="_blank" rel="noopener">issue</a>
```
Comment (Eatham532) [inline mkdocs.yml:263]: same thing here with the repo

**Comment by Eatham532** [comment]
Are you done yet? If not can you set this as a draft and then set this as ready for review when finished please

**Comment by haleema-khatun** [comment]
I am done , I have already raised a PR , it's being two days.

File: mkdocs.yml [hunk 3]
```diff
@@ -153,13 +163,23 @@ markdown_extensions:
           class: js-template
           format: !!python/name:extensions.code_fence_formatter.format_template
           validator: !!python/name:extensions.code_fence_formatter.fence_validator
+        - name: mermaid
```
Comment (Eatham532) [inline mkdocs.yml:166]: The project doesn't use mermaid fence blocks. Instead it uses kroki. You don't need this line.

File: mkdocs.yml [hunk 4]
```diff
@@ -68,27 +66,39 @@ theme:
       accent: indigo
       toggle:
         icon: material/brightness-4
-        name: Switch to system preference
-  
+        name: Switch to light mode
```
Comment (Eatham532) [inline mkdocs.yml:69]: I'm not too sure why you changed this label. The cycle for changing themes goes 

light --> Dark --> Automatic.

So when in dark mode, you want the button to be a toggle for system preferences.

File: mkdocs.yml [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-site_url: https://eatham532.github.io/Software-Engineering-HSC-Textbook/
-repo_name: Eatham532/Software-Engineering-HSC-Textbook
+repo_url: https://github.com/haleema-khatun/Software-Engineering-HSC-Textbook/
+site_url: https://haleema-khatun.github.io/Software-Engineering-HSC-Textbook/
+repo_name: haleema-khatun/Software-Engineering-HSC-Textbook
 edit_uri: edit/main/docs/
 
 copyright: >
-  Copyright &copy; 2025 Eatham532 –
+  Copyright &copy; 2024 Haleema Khatun –
```
Comment (Eatham532) [inline mkdocs.yml:10]: You changed the copyright as well? And for 2024?

File: mkdocs.yml [hunk 6]
```diff
@@ -1,13 +1,13 @@
-site_name: Software Engineering Textbook HSC
-site_description: Comprehensive educational resource for NSW HSC Software Engineering syllabus
+site_name: NSW HSC Software Engineering Textbook
```
Comment (Eatham532) [inline mkdocs.yml:1]: Why did you change the textbook name? It used to be this, but I wanted the name to start with what it is not what it is for.

File: mkdocs.yml [hunk 7]
```diff
@@ -168,8 +188,15 @@ markdown_extensions:
...
-    
-      
+  - pymdownx.tabbed:
+      alternate_style: true
+      slugify: !!python/object/apply:pymdownx.slugs.slugify
+        kwds:
+          case: lower
+  - pymdownx.tasklist:
+      custom_checkbox: true
+  - pymdownx.tilde
```
Comment (Eatham532) [inline mkdocs.yml:198]: This extension isn't used in the project...

File: mkdocs.yml [hunk 8]
```diff
@@ -180,13 +207,17 @@ extra_css:
...
 
 extra_javascript:
   - assets/diagram-modal.js
   - assets/quiz.js
   - assets/ide-utils.js
   - assets/code-blocks.js
   - assets/glossary.js
+  - javascripts/mathjax.js
+  - https://polyfill.io/v3/polyfill.min.js?features=es6
+  - https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js
```
Comment (Eatham532) [inline mkdocs.yml:220]: What are these scripts for?

File: mkdocs.yml [hunk 9]
```diff
@@ -153,13 +163,23 @@ markdown_extensions:
...
   - pymdownx.escapeall
   - toc:
       permalink: true
+      toc_depth: 3
   - pymdownx.highlight:
       anchor_linenums: true
       line_spans: "__span"
       pygments_lang_class: true
+  - pymdownx.arithmatex:
+      generic: true
```
Comment (Eatham532) [inline mkdocs.yml:178]: not sure why this has anything related to previous and next buttons... I don't have math being used in this documentation

File: mkdocs.yml [hunk 10]
```diff
@@ -153,13 +163,23 @@ markdown_extensions:
...
+  - pymdownx.betterem:
+      smart_enable: all
+  - pymdownx.caret
+  - pymdownx.details
```
Comment (Eatham532) [inline mkdocs.yml:182]: we don't use pymdownx.details. Instead we use pymdownx.blocks.details. It has better syntax that makes it easier to maintain the project.

File: docs/stylesheets/extra.css
Comment (Eatham532) [inline docs/stylesheets/extra.css:1]: Image

Hovering adds this weird curve. Is this intended? I would probably just remove the border radius entirely

Image

I see that you are trying to show the users position with the left border. But the border style is kind of an eye sore. Maybe just change the background colour instead.

Image

You can't tell the difference between these two "headers"  Maybe don't put a highlight on the primary header (The year 11) and instead add a light underline.

File: mkdocs.yml [hunk 11]
```diff
@@ -68,27 +66,39 @@ theme:
...
+    - navigation.path
+    - navigation.indexes
+    - navigation.footer          # ← CRITICAL: ENABLES PREV/NEXT BUTTONS
     - search.highlight
     - search.suggest
+    - search.share
     - content.code.annotate
     - content.code.copy
     - content.code.select
-    - content.tabs.link
```
Comment (Eatham532) [inline mkdocs.yml:85]: Why did you remove this? This feature is important to have tab sync for code blocks.

File: mkdocs.yml [hunk 12]
```diff
@@ -68,27 +66,39 @@ theme:
...
     - content.action.view
     - toc.follow
+    - toc.integrate
 
+  font:
+    text: Roboto
+    code: Roboto Mono
 
 markdown_extensions:
+  - abbr
```
Comment (Eatham532) [inline mkdocs.yml:101]: This is handled by the glossary tooltip feature. Don't add this extension. I am open to considering using if it improves things. right now it isn't needed though

File: mkdocs.yml [hunk 13]
```diff
@@ -68,27 +66,39 @@ theme:
...
     - navigation.prune
+    - navigation.tracking
+    - navigation.sections
+    - navigation.expand
+    - navigation.path
+    - navigation.indexes
+    - navigation.footer          # ← CRITICAL: ENABLES PREV/NEXT BUTTONS
     - search.highlight
     - search.suggest
+    - search.share
```
Comment (Eatham532) [inline mkdocs.yml:87]: This feature is kinda stupid in my opinion. Who wants to share their search? Remove this.

File: docs/stylesheets/extra.css
Comment (Eatham532) [reply docs/stylesheets/extra.css:1]: Image

Previous and Next buttons aren't even found on the page.... this is why I thought the PR wasn't complete

File: mkdocs.yml [hunk 14]
```diff
@@ -68,27 +66,39 @@ theme:
...
     - content.code.annotate
     - content.code.copy
     - content.code.select
-    - content.tabs.link
     - content.action.edit
     - content.action.view
     - toc.follow
+    - toc.integrate
```
Comment (Eatham532) [inline mkdocs.yml:94]: toc.integrate doesn't show the current page's table of contents. Which is it's primary use. What was wrong with the old method separating navigation of the textbook with the navigation of the page?

Because pages are so long, It is really hard to find things when everything is on one long scroll "page"

File: docs/stylesheets/extra.css
Comment (Eatham532) [reply docs/stylesheets/extra.css:1]: Be more specific with the naming of "extra.css" or just move it into common. (That is the file where you just have generic styles that are found everywhere in the textbook.

File: mkdocs.yml [hunk 15]
```diff
@@ -168,8 +188,15 @@ markdown_extensions:
       slugify: !!python/object/apply:pymdownx.slugs.slugify
         kwds:
           case: lower
-    
-      
+  - pymdownx.tabbed:
+      alternate_style: true
+      slugify: !!python/object/apply:pymdownx.slugs.slugify
+        kwds:
+          case: lower
```
Comment (Eatham532) [inline mkdocs.yml:195]: Why did you add this extension when we already had pymdownx.blocks.tab which is the better version and the one that is actually being used?

**Review by Eatham532 — CHANGES_REQUESTED** [review]
Thanks for the work that you have done! I really appreciate the time you are putting to to help. However I think you need to rethink a few of the changes you made. It appears that you made a lot more changes that isn't related to the goal of this PR. 

Please try to outline exactly what is being changed and why. And if you do want to make multiple other changes please try to separate those into separate PR's. Thanks :)

**Comment by haleema-khatun** [comment]
My bad ,  I'll try to fix all these changes and then raise the PR.
Thanks for your guidance 😃
